hack/update-deps.sh

```diff
@@ -61,3 +61,8 @@ find vendor/ -name '*_test.go'-delete
 export GOFLAGS=-mod=vendor
 
 update_licenses third_party/VENDOR-LICENSE "./..."
+
+
+# Patch k8s leader-election fixing graceful release
+# More information: https://github.com/kubernetes/kubernetes/pull/91942
+patch vendor/k8s.io/client-go/tools/leaderelection/leaderelection.go hack/k8s-client-go.patch
```

Comment:
Historically we've used `git apply` to avoid introducing new tooling dependencies here.  This is making the auto-deps automation fail :(